docs/DeveloperGuide.md

```diff
@@ -173,48 +173,58 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
-### Show assignment list feature
+### Assignment list panel display feature
 
 #### Implementation
 
-The show assignment mechanism is facilitated by `AddressBook`, where the specified person's assignment list is stored internally under `assignments` This `assignments` is retrieved or updated by the following methods:
+The assignment list panel display mechanism is facilitated by `AddressBook`, where the specified person's assignment list is stored internally under `assignments` This `assignments` is retrieved or updated by the following methods:
 * `AddressBook#getAssignmentList()`
 * `AddressBook#updateAssignmentList(Person person)`  —  where `person` is the specified person.
 
-These methods are exposed in the `Model` interface as `Model#getFilteredAssignmentList()` and `Model#updateFilteredAssignmentList(Person person)` respectively.
+These methods are exposed in the `Model` interface as `Model#getAssignmentList()` and `Model#updateAssignmentList(Person person)` respectively.
+In addition, `Model` contains a field `assignmentsList` that points towards `assignment` in `AddressBook`. `assignmentsList` serve as connection for `Logic` to retrieve `assignments` to display.
 
 Given below is an example usage scenario and how the show assignment mechanism behaves at each step.
 
-Step 1. The user launches the application for the first time. The `assignments` will be initialized with an empty list.
+Step 1. The user launches the application for the first time. The `assignments` will be initialized with a `UniqueAssignmentList` that does not contain any `Assignment`.
 
-![AssignmentState0](images/AssignmentState0.png)
+Step 2. The user inputs `show 2` command to display the 2nd person's assignment list in the address book. The `show` command will then call `Model#FilterdAssignmentList(person)`, whereby `person` variable is the 2nd person in the address book.
+This will then call `Addressbook#updateAssignmentList(person)`, causing the `assignments` in `AddressBook` to be replaced with the assignments in 2nd person's assignment list.
 
-Step 2. The user inputs `show 2` command to display the 2nd person's assignment list in the address book. The `show` command will then call `Model#updateFilterdAssignmentList(person)`, whereby `person` variable is the 2nd person in the address book.
-This causes the `assignments` in `AddressBook` to be replaced with the 2nd person's assignment list.
+The two object diagram below shows illustrates how the objects interacts and changes when a `show` command is executed.
 
-![AssignmentState1](images/AssignmentState1.png)
+![DisplayAssignmentObjectDiagram1](images/DisplayAssignmentObjectDiagram1.png))
+![DisplayAssignmentObjectDiagram2](images/DisplayAssignmentObjectDiagram2.png))
 
+Step 3. When `assignments` is updated, it is retrieved by the `Logic` using `Model#getAssignmentList()` to input into the assignment list panel of the `UI`
```

Comment:
Step 3. When `assignments` is updated, the assignment list panel of the `Ui` will be updated accordingly since it an observer of the `assignments` list in `Model`.